models/Asset/Thumbnail/ImageThumbnailTrait.php

```diff
@@ -245,6 +245,16 @@ public function getDimensions()
                 $dimensions = $this->readDimensionsFromFile();
             }
 
+            if ($config && $config->getHighResolution() && $config->getHighResolution() > 1) {
+                if(!empty($dimensions['width'])) {
+                    $dimensions['width'] = (int)floor($dimensions['width'] / $config->getHighResolution());
+                }
+
+                if (!empty($dimensions['height'])) {
+                    $dimensions['height'] = (int)floor($dimensions['height'] / $config->getHighResolution());
+                }
```

Comment:
LGTM, the only thing is that could be refactored a ltitle bit:

here its divided by the high resolution and later on it gets multiplied "back" for it, see:
https://github.com/pimcore/pimcore/blob/a6726e031e9fe99d12841dea1283cd9964d4e26c/models/Asset/Thumbnail/ImageThumbnailTrait.php#L265-L268
also the `!empty` looks something that should be checked once, later on it would overkill a little bit
https://github.com/pimcore/pimcore/blob/a6726e031e9fe99d12841dea1283cd9964d4e26c/models/Asset/Thumbnail/ImageThumbnailTrait.php#L258-L259


suggesting to refactor a little bit, something like:

```
$this->width = $this->realWidth = (int)$dimensions['width'] ?? null;
$this->height = $this->realHeight = (int)$dimensions['height'] ?? null;

if ($config && $config->getHighResolution() && $config->getHighResolution() > 1) {
    $this->width = (int)floor($this->realWidth / $config->getHighResolution());
    $this->height = (int)floor($this->realHeight / $config->getHighResolution());
}
```

Reply:
@BlackbitDevs any update on this? should i take over and push the changes?

Reply:
ahh ok, no problem 😄 thank you!